### PR TITLE
Refactoring front end python code

### PR DESCRIPTION
**Pull request by ajyl**
Refactor all front-end functionalities. Has dependencies on #16 , will need a rebase.

### CODE REVIEW

File: potato/flask_server.py [hunk 1]
```diff
@@ -3827,6 +2544,7 @@ def run_create_task_cli():
     """
     Run create_task_cli().
     """
+    import webbrowser
```
Comment (davidjurgens) [inline potato/flask_server.py:2547]: what's with the webbrowser import here?

**Review by davidjurgens — CHANGES_REQUESTED** [review]
In general this looks pretty good! The flask_server.py was certainly getting overly long. Aside from the `import webbrowser`, I'm happy to approve this change. 

I do have some weird lingering concern that there's this logical coupling in the span annotation generation where it has to set the span colors there and that's happening in the `config` which feels back-end. I'm not sure how to refactor that exactly and it probably shouldn't be a part of this pull but it does seem like a candidate for a future refactor.

The schemas could all also use a refactor to clean out the common cruft like setting keybindings into basic functionality. Also best left to a later refactor once this is merged.

File: potato/flask_server.py (hunk 1 above)
Comment (ajyl) [reply potato/flask_server.py:2547]: I refactored `main()` to either call `run_create_task_cli()` or `run_server()` -- my understanding is that `webbrowser` is only being used for `create_task_cli()`. I can move it back to importing at a module level.

Comment (davidjurgens) [reply potato/flask_server.py:2547]: Oh right! I think we could actually remove that import and the code that uses it and replace it with a `raise Exception("Gui-based design not implemented yet")` kind of message. I think this pull request is all good to go then.

**Comment by ajyl** [comment]
Once #16 gets in, I can rebase this branch and I think it'll be ready for merge as well.
There should be no change in functionality, and I did some ad-hoc testing, but there's a chance some things will break... I started a PR to start building unittests in another PR (#18). 

Some of the functionality might have been broken before this PR, however. For instance [this line](https://github.com/davidjurgens/potato/blob/master/potato/flask_server.py#L3546) seems to be a bug, my guess is that `label_data` got copy-pasted somewhere. I'm not sure what the correct functionality should be so this bug is not being resolved in this [PR](https://github.com/davidjurgens/potato/pull/17/files#diff-3a63009c66e7d4a0ae8dc410f739de2753b5cc93feb5bca5bbd0e3c52ee5afcdR52). I agree that another round of refactoring each schema-function should remove any additional bugs that may be hiding.

**Comment by davidjurgens** [comment]
Ok, I've merged in #16 so this is ready for a rebase so we can merge this pull request

**Comment by ajyl** [comment]
Refactored!

**Review by davidjurgens — APPROVED** [review]

**Comment by davidjurgens** [comment]
Merging in the new GUI front-end changes

**Pull request merged**